static_analyzer/__init__.py

```
@@ -285,6 +286,59 @@ def get_adapter_for_file(self, file_path: Path) -> tuple[LanguageAdapter, Path]
+
+            queries = [(file_path, line, char) for line, char in call_sites]
+            results, _ = client.send_definition_batch(queries)
+
+            resolved = file_path.resolve()
+            unique_paths: set[str] = set()
+            for definitions in results:
+                for defn in definitions:
+                    uri = defn.get("uri", "")
+                    if not uri.startswith("file://"):
+                        continue
+                    dep_path_obj = uri_to_path(uri)
```

Ruff (0.15.9)

[error] Undefined name `uri_to_path`

(F821)